tests/minus.rs

```diff
@@ -0,0 +1,18 @@
+#[cfg(test)]
+mod tests {
+    use lrlex::lrlex_mod;
+    use lrpar::lrpar_mod;
+    use yaiwr::eval;
+    lrlex_mod!("calc.l");
+    lrpar_mod!("calc.y");
+    #[test]
+    fn eval_minius_exp() {
+        let lexerdef = calc_l::lexerdef();
+        let input = "1-1".to_string();
+        let result = eval(&lexerdef, &input);
+        match result {
+            Ok(msg) => assert!(msg == "Result: 0"),
+            _ => {}
```

Comment:
Shouldn't the `err` case panic?

You could do something more concise with `matches!`.

Reply:
Similarly in other places.

Reply:
adedd panic!
can you give an example of how would you use `matches!` here? I'll read more about that macro

Reply:
glad you commented on `panic!`,  I had a test for `-` operation that doesnt exist in this calc implementation, adding `panic!` helped with identifying that :)

Reply:
> can you give an example of how would you use matches!

something like:

```
assert!(matches!(result, Ok("Result: 0")))
```

I'm not 100% if it will work, because presumably it's a `String` inside and not a `&str`. Worth a try though.

Reply:
Tried it, looks like I can assert on the enum option but not on the value of the string.
So something like this:
```rust
let lexerdef = calc_l::lexerdef();
let res = eval(&lexerdef, "2*2");
let  _expected = "YOU SHALL NOT PASS";
matches!(res, Result::Ok(_expected)));
```
will pass the assertion

Reply:
Did you forget to wrap the `matches!` in an `assert!` though?

Reply:
ah, you're right I forgot to add it to the example I gave, should be:
```
....
assert!(matches!(res, Result::Ok(_expected)));
```

Reply:
https://replit.com/@ihcmikmai/matchesmacro#src/main.rs

Reply:
You're right, it doesn't work as expected.

OK, I'd do something like this instead then:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=42a96639daf88f15a610e60d239d89cd

works as long as all constituent types implement `PartialEq`.

Reply:
done